src/global.cpp

```diff
@@ -88,6 +93,61 @@ std::shared_future<void> Cleanup() { return impl::Init::Instance().cleanup(); }
 
 void SetSctpSettings(SctpSettings s) { impl::Init::Instance().setSctpSettings(std::move(s)); }
 
+std::map<int, UnhandledStunRequestHandler *> unboundStunCallbacks;
+
+#if !USE_NICE
+
+void InvokeUnhandledStunRequestCallback (const juice_mux_binding_request *info, void *user_ptr) {
+	PLOG_DEBUG << "Invoking unhandled STUN request callback";
+
+	UnhandledStunRequestHandler *handler = (struct UnhandledStunRequestHandler*)user_ptr;
+
+	if (handler->callback) {
+		handler->callback({
+			std::string(info->local_ufrag),
+			std::string(info->remote_ufrag),
+			std::string(info->address),
+			info->port
+		}, handler->userPtr);
+	} else {
+		PLOG_DEBUG << "No unhandled STUN request callback configured for port " << info->port;
+	}
+}
+
+#endif
+
+void OnUnhandledStunRequest ([[maybe_unused]] std::string host, [[maybe_unused]] int port, [[maybe_unused]] UnhandledStunRequestCallback callback, [[maybe_unused]] void *userPtr) {
```

Comment:
The name should reflect that this listens only on mux mode, and that it attempts to listen to a specific port, you could rename it to `ListenIceUdpMux()` for instance, since the feature is called `enableIceUdpMux` in `rtc::Configuration`.

`host` should also be optional to allow any address (it would pass NULL to libjuice), and you could remame it to `bindAddress` as it is the name in the configuration. If it's not part of the mapping, I think it should be moved at the end and made an optional parameter.

Reply:
Fixed!

Reply:
> `host` should also be optional to allow any address (it would pass NULL to libjuice)

I guess the only problem with this is the invocation to unbind a listener with a host now looks a bit awkward, certainly in js:

```js
listenIceUdpMux(port, null, host)
```

How about passing a `bind_address` struct with `host`/`port` fields as the first argument instead?  That way the host field can be omitted if it should bind to all available addresses.

Reply:
> How about passing a `bind_address` struct with `host`/`port` fields as the first argument instead? That way the host field can be omitted if it should bind to all available addresses.

Good idea, it could be something like that, on the model of `WebSocketServerConfiguration`:
```
struct IceUdpMuxListenerConfiguration {
    uint16_t port;
    optional<string> bindAddress;
}
```